core/src/main/java/jenkins/model/RunIdMigrator.java

```diff
@@ -350,6 +350,11 @@ private void unmigrateJobsDir(File jobs) throws Exception {
             return;
         }
         for (File job : jobDirs) {
+
+            if (job.getName().equals("builds")) {
```

Comment:
Comment now on line 369 should be edited to clarify that the older clause handles grandchildren from `getRootDirFor` whereas this clause handles direct children.